action/protocol/staking/protocol.go

```diff
@@ -601,12 +602,14 @@ func (p *Protocol) settleAction(
 	return &r, nil
 }
 
-func (p *Protocol) needToReadCandsMap(height uint64) bool {
-	return height > p.config.PersistStakingPatchBlock
+func (p *Protocol) needToReadCandsMap(ctx context.Context, height uint64) bool {
+	fCtx := protocol.MustGetFeatureWithHeightCtx(ctx)
+	return height > p.config.PersistStakingPatchBlock && !fCtx.FixCandCenterAlias(height)
```

Comment:
why we have `PersistStakingPatchBlock` from p.config, but the other from `fCtx`?

Reply:
`PersistStakingPatchBlock` is not a hard-fork height (so not in genesis), while `FixCandCenterAlias()` is according to the next hard-fork height in genesis